src/com/maddyhome/idea/vim/helper/SearchHelper.java

```diff
@@ -119,12 +120,12 @@ public static TextRange findBlockRange(@NotNull Editor editor, char type, int co
     int loc = blockChars.indexOf(type);
     char close = blockChars.charAt(loc + 1);
 
-    int bstart = findBlockLocation(chars, close, type, -1, pos, count);
+    int bstart = findBlockLocation(chars, close, type, -1, pos, count, checkMatchForChar);
```

Comment:
This will skip `"` analysis and miss cases like: `{"{foo, <caret>bar", baz}}` that should remain intact after `di{`.

Also it will ignore `'` so `{'{foo, 'bar', baz}'}` after `di{` won't become `{'{}'}`.